app/main.py

```diff
@@ -2,0 +1,43 @@
+from __future__ import annotations
+
+
+class Animal:
+
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        if self.health >= 0:
+            self.alive = True
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        if self.hidden:
+            self.hidden = False
+        else:
+            self.hidden = True
+
+
+class Carnivore(Animal):
+
+    def bite(self, other: Animal) -> None:
+        if isinstance(other, Herbivore) and not other.hidden:
+            other.health -= 50
+        if other.health <= 0:
+            other.alive = False
+            Animal.alive.remove(other)
```

Comment:
Removing elements from the `alive` list while iterating over it can lead to unexpected behavior. Ensure that this operation is safe or consider using a different approach to handle dead animals.